### PR TITLE
feat(api): add String-based API support

### PR DESCRIPTION
**Pull request by innomaxx**

### CODE REVIEW

**Review by andrii-bodnar — COMMENTED** [review]
Branch clone and merge methods - https://developer.crowdin.com/api/v2/string-based/#tag/Branches

File: src/Crowdin.Api/SourceStrings/SourceStringsApiExecutor.cs [hunk 1]
```diff
@@ -126,6 +126,30 @@ public async Task<SourceString> EditString(int projectId, int stringId, IEnumera
             CrowdinApiResult result = await _apiClient.SendPatchRequest(url, patches);
             return _jsonParser.ParseResponseObject<SourceString>(result.JsonObject);
         }
+        
+        /// <summary>
+        /// Upload strings status. Documentation:
+        /// <a href="https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.strings.uploads.get">Crowdin API</a>
```
Comment (andrii-bodnar) [inline src/Crowdin.Api/SourceStrings/SourceStringsApiExecutor.cs:132]: There is also Enterprise SB doc - https://developer.crowdin.com/enterprise/api/v2/string-based/

File: src/Crowdin.Api/Branches/Branch.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+using System;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+
+namespace Crowdin.Api.Branches
+{
+    [PublicAPI]
+    public class Branch
```
Comment (andrii-bodnar) [inline src/Crowdin.Api/Branches/Branch.cs:9]: We already have the `Branch` class - `src/Crowdin.Api/SourceFiles/Branch.cs`

File: src/Crowdin.Api/Branches/BranchesApiExecutor.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            return _jsonParser.ParseResponseObject<BranchCloneStatus>(result.JsonObject);
+        }
+        
+        /// <summary>
+        /// List Branches. Documentation:
+        /// <a href="https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.branches.getMany">Crowdin API</a>
+        /// <a href="https://developer.crowdin.com/enterprise/api/v2/string-based/#operation/api.projects.branches.getMany">Crowdin Enterprise API</a>
+        /// </summary>
+        [PublicAPI]
+        public async Task<ResponseList<Branch>> ListBranches(
```
Comment (andrii-bodnar) [inline src/Crowdin.Api/Branches/BranchesApiExecutor.cs:76]: Some of these branch methods are already included in the source files namespace, we need to keep the existing without duplicating

File: tests/Crowdin.Api.Tests/Branches/BranchesCrudApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+using Crowdin.Api.Branches;
+using Crowdin.Api.Core;
+using Crowdin.Api.Tests.Core;
+
+namespace Crowdin.Api.Tests.Branches
+{
+    public class BranchesApiTests
```
Comment (andrii-bodnar) [inline tests/Crowdin.Api.Tests/Branches/BranchesCrudApiTests.cs:19]: Duplicates the existing tests

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]

File: src/Crowdin.Api/Branches/BranchesSortingRule.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+
+using System.ComponentModel;
+using JetBrains.Annotations;
+
+namespace Crowdin.Api.Branches
+{
+    [PublicAPI]
+    public enum BranchesSortingRule
```
Comment (andrii-bodnar) [inline src/Crowdin.Api/Branches/BranchesSortingRule.cs:8]: It looks like this does not support complex ordering like `orderBy=createdAt desc,name,priority`

File: src/Crowdin.Api/Branches/Branch.cs (hunk 2 above)
Comment (innomaxx) [reply src/Crowdin.Api/Branches/Branch.cs:9]: discussed

Comment (andrii-bodnar) [inline src/Crowdin.Api/Branches/Branch.cs:9]: Can we extend the source files branch model from this model?

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**